dashboard/app/models/pd/workshop_survey.rb

```diff
@@ -127,15 +121,66 @@ def validate_required_fields
 
     # if this is the first survey completed by this user, also require
     # demographics questions
-    if pd_enrollment && self.class.find_by_user(pd_enrollment.user).empty?
-      demographics_required_fields.each do |field|
-        add_key_error(field) unless hash.key?(field)
+    if pd_enrollment
+      if pd_enrollment.user.nil? || self.class.find_by_user(pd_enrollment.user).empty?
+        demographics_required_fields.each do |field|
+          add_key_error(field) unless hash.key?(field)
+        end
       end
     end
 
     super
   end
 
+  # Simple helper that iterates over each facilitator as reported by the user
+  # and each facilitator-specific field and yields a block with the facilitator,
+  # the field, and the combined field name we expect in the flattened version of
+  # our hash
+  def each_facilitator_field(hash=nil)
+    hash ||= sanitize_form_data_hash
+
+    facilitators = hash.try(:[], :who_facilitated) ||
+      hash.try(:[], 'whoFacilitated') ||
```

Comment:
this is fine, but out of curiosity why might it be in both forms? Should we normalize with `.symbolize_keys` or `.stringify_keys`?